src/fdosecrets/widgets/SettingsModels.cpp

```diff
@@ -344,48 +344,49 @@ namespace FdoSecrets
         }
     }
 
-    void SettingsSessionModel::populateModel()
+    void SettingsConnectionModel::populateModel()
     {
         beginResetModel();
 
-        m_sessions.clear();
+        m_connections.clear();
 
         if (m_service) {
             // Add existing database tabs
-            for (const auto& sess : m_service->sessions()) {
-                sessionAdded(sess, false);
+            for (const auto& conn : m_service->connections()) {
+                connectionAdded(conn, false);
             }
 
             // connect signals
-            connect(m_service, &Service::sessionOpened, this, [this](Session* sess) { sessionAdded(sess, true); });
-            connect(m_service, &Service::sessionClosed, this, &SettingsSessionModel::sessionRemoved);
+            connect(
+                m_service, &Service::connectionOpened, this, [this](Connection* conn) { connectionAdded(conn, true); });
+            connect(m_service, &Service::connectionClosed, this, &SettingsConnectionModel::connectionRemoved);
         }
 
         endResetModel();
     }
 
-    void SettingsSessionModel::sessionAdded(Session* sess, bool emitSignals)
+    void SettingsConnectionModel::connectionAdded(Connection* sess, bool emitSignals)
     {
-        int row = m_sessions.size();
+        int row = m_connections.size();
         if (emitSignals) {
             beginInsertRows({}, row, row);
         }
 
-        m_sessions.append(sess);
+        m_connections.append(sess);
 
         if (emitSignals) {
             endInsertRows();
         }
     }
 
-    void SettingsSessionModel::sessionRemoved(Session* sess)
+    void SettingsConnectionModel::connectionRemoved(Connection* sess)
     {
-        for (int i = 0; i != m_sessions.size(); i++) {
-            if (m_sessions[i] == sess) {
+        for (int i = 0; i != m_connections.size(); i++) {
+            if (m_connections[i] == sess) {
                 beginRemoveRows({}, i, i);
 
-                m_sessions[i]->disconnect(this);
-                m_sessions.removeAt(i);
+                m_connections[i]->disconnect();
```

Comment:
This `disconnect` is only for setting disconnected state in the `Connection` object, but not actually disconnects signals from the connection to this model, i.e. `QObject::disconnect`. See also the comments on `Connection::disconnect`.